scripts/fetch/view_limits.R

```diff
@@ -0,0 +1,90 @@
+#' @title Fetch viz view limits
+#' @param viz a vizlab object including a \code{parameter_spatial} parameter input.
+#' @details \code{parameter_spatial} must include:
+#' /describe{
+#'   \item{id}{parameter_spatial}
+#'   \item{crs}{valid crs for \pkg{sf}}
+#'   \item{bbox}{numeric in xmin, ymin, xmax, ymax order}
+#'   \item{height}{figure height inches}
+#'   \item{width}{figure width inches}
+#'   \item{pointsize}{number of pixels per inch}
+#' }
+fetch.view_limits <- function(viz = as.viz('fetch_view_limits')){
+  
+  deps <- readDepends(viz)
+  required <- "parameter_spatial"
+  checkRequired(deps, required)
+  p_spatial <- deps[["parameter_spatial"]]
+  checkRequired(p_spatial, c("bbox", "width", "height", "pointsize"))
+    
+  bbox_polygon <- bbox_to_polygon(p_spatial$bbox, 
+                            return_crs = p_spatial$crs)
+  
+  view_limits <- plot_view_limits(bbox_polygon,
+                               width = p_spatial$width, 
+                               height = p_spatial$height, 
+                               pointsize = p_spatial$pointsize)
+  
+  saveRDS(view_limits, viz[['location']])
+}
+
+fetchTimestamp.view_limits <- neverCurrent
+
+#' @title Construct sf polygon box from two corner numeric bbox.
+#' @param bbox numeric xmin, ymin, xmax, ymax in WGS84 (EPSG:4326) lon,lat. 
+#' No rearranging is done!
+#' @param return_crs if the returned bbox should be in a projection other than EPSG:4326, 
+#' pass in a crs string compatible with sf::st_transform()
+#' 
+#' @return an object of class 'bbox' from sf::
+#' @example 
+#' bbox <- construct_sf_bbox(c(-87, 21, -70, 34), return_crs = "+init=epsg:5070")
+bbox_to_polygon <- function(bbox, return_crs = NULL) {
+  bbox_poly <- sf::st_sfc(sf::st_polygon(list(matrix(c(bbox[c(1,2)], 
+                                  bbox[c(1,4)], 
+                                  bbox[c(3,4)], 
+                                  bbox[c(3,2)], 
+                                  bbox[c(1,2)]), 
+                                ncol = 2, byrow = T)), 
+                    dim = "XY"), 
+         crs = "+init=epsg:4326")
```

Comment:
this could be simpler:
```
bbox_poly <- sf::st_sfc(sf::st_polygon(list(
      matrix(
        bbox[c(1,2,
               1,4,
               3,4,
               3,2,
               1,2)], 
        ncol = 2, byrow = T)), 
    dim = "XY"), 
    crs = "+init=epsg:4326")
```